### PR TITLE
Fix Makefile for Travis and Miniconda

### PR DESCRIPTION
**Pull request by ShigekiKarita**
This PR addresses several issues in `tools/Makefile`
1. system python is default. miniconda is more portable and better
1. travis does not use `Makefile` but it does repeat its lines
1. testing commands (pytest, flake8, autopep8) are too complex for local develop. they should be done by `make test`
1. we need an option to skip building kaldi https://github.com/espnet/espnet/issues/14#issuecomment-369886968

### CODE REVIEW

**Comment by kan-bayashi** [comment]
I will try to test from now.

**Comment by kan-bayashi** [comment]
`Make` failed. 
``` 
$ make
make: *** No rule to make target `kaldi', needed by `all'.  Stop.
```

File: tools/Makefile [hunk 1]
```diff
@@ -1,51 +1,109 @@
...
+### kaldi related ###
 kaldi-io-for-python:
 	git clone https://github.com/vesis84/kaldi-io-for-python.git
 	cd ../src/utils; ln -s ../../tools/kaldi-io-for-python/kaldi_io.py kaldi_io_py.py
 
-kaldi: kaldi_github
-	cd kaldi_github/tools; $(MAKE) all
-	cd kaldi_github/src; ./configure --shared --use-cuda=no; $(MAKE) depend; $(MAKE) all
-	ln -s kaldi_github kaldi
+kaldi/: $(KALDI_BIN) $(BEAMFORMIT)
```
Comment (kan-bayashi) [inline tools/Makefile]: `kaldi/` -> `kaldi`

**Comment by kan-bayashi** [comment]
Conda requires to install all of the libraries such as cuda, and it forces to use it. 
(Basically, these libraries are install in root directory.)
Therefore, it causes unexpected behaviors in several times, so I dislike conda.
Is it possible to select whether or not to use conda?

**Comment by ShigekiKarita** [comment]
OK. so I will add flag to select pip or conda to install packages.

And I recommend to use conda for venv because system python is not uniform
e.g. ubuntu has python 2.7.14 centos has 2.7.5

2018年3月29日(木) 17:07 Tomoki Hayashi <notifications@github.com>:

**Comment by ShigekiKarita** [comment]
I will update .travis.yml for 4 cases
- conda 2.7
- conda 3.6
- system 2.7
- system 3.6

**Comment by kan-bayashi** [comment]
@ShigekiKarita Thank you, Shigeki.
BTW, I got following error in kaldi install with miniconda.
```
Successfully installed Cython-0.28.1 chainer-3.5.0 filelock-3.0.4 protobuf-3.5.2.post1 python-speech-features-0.6 setuptools-39.0.1
. venv/bin/activate && cd kaldi_github/src; ./configure --shared --use-cuda=no --mathlib=MKL --mkl-libdir=../../venv/lib --mkl-root=../../venv; make depend; make all
Configuring KALDI to use MKL
Configuring KALDI to use MKL
Configuring ...
Checking compiler g++ ...
Checking OpenFst library in  ...
***configure failed: Could not find file /include/fst/fst.h:
  you may not have installed OpenFst. See ../tools/INSTALL ***
make[1]: Entering directory `/work4/t_hayashi/work/espnet-fix-make/tools/kaldi_github/src'
Makefile:30: kaldi.mk: No such file or directory
kaldi.mk does not exist; you have to run ./configure
make[1]: *** [kaldi.mk] Error 1
make[1]: Leaving directory `/work4/t_hayashi/work/espnet-fix-make/tools/kaldi_github/src'
make[1]: Entering directory `/work4/t_hayashi/work/espnet-fix-make/tools/kaldi_github/src'
Makefile:30: kaldi.mk: No such file or directory
kaldi.mk does not exist; you have to run ./configure
make[1]: *** [kaldi.mk] Error 1
make[1]: Leaving directory `/work4/t_hayashi/work/espnet-fix-make/tools/kaldi_github/src'
make: *** [kaldi_github/src/feat/libkaldi-feat.so] Error 2
```

**Comment by sw005320** [comment]
Sorry that I didn't follow so much. This seems to be a big change (but I like a big change). still several people have issues in the installation, and I appreciate your efforts. Please keep on working.

**Comment by sw005320** [comment]
What is the status of this?

**Pull request closed without merging**